src/app/lib/getQuestBySlug.ts

```diff
@@ -18,5 +33,12 @@ export async function getQuestBySlug(slug: string) {
 
   const data = await res.json(); // Extract data from the response
 
-  return { data, url: apiBaseUrl }; // Return a plain object
+  if (!data || !Array.isArray(data.data) || data.data.length === 0) {
```

Comment:
This part seems weird and does not seem to achieve anything, let's have a talk about it